src/main/java/com/scaleunlimited/flinkcrawler/topology/CrawlTopologyBuilder.java

```diff
@@ -399,6 +422,26 @@ public FetchUrl map(Tuple3<CrawlStateUrl, FetchUrl, FetchUrl> justHasFetchUrl)
                                 TimeUnit.SECONDS, 10000)
                         .name("FetchUrlsFunction");
 
+        // Save off the content by converting it to a WARC record and passing it on to the provided
+        // content sink function.
+        DataStream<FetchResultUrl> fetchResultUrlsToSave = fetchResultUrls;
+        SingleOutputStreamOperator<Tuple2<NullWritable, WARCWritable>> warcStream = fetchResultUrlsToSave
+                .flatMap(new CreateWARCWritableFunction(_userAgentString))
+                .name("Create WARC writable");
+        
+        DataStreamSink<Tuple2<NullWritable, WARCWritable>> contentSinkUsingOutputFormat = null;
+        if (_contentSink != null) {
+            contentSinkUsingOutputFormat = warcStream.addSink(_contentSink);
+        } else {
+            Job job = Job.getInstance();
+            WARCOutputFormat.setOutputPath(job, new Path(_contentFilePathString));
+            HadoopOutputFormat<NullWritable, WARCWritable> hadoopOutputFormat = new HadoopOutputFormat<NullWritable, WARCWritable>(new WARCOutputFormat(), job);
```

Comment:
It's strange that you don't seem to be using your WARCOutputFormat (except for that one static? method call above), but rather HadoopOutputFormat.

Reply:
Actually it is being used above in addition to the static method call 

    new HadoopOutputFormat<NullWritable, WARCWritable>(new WARCOutputFormat(), job);

Reply:
OK